source/programming/python.md

```diff
@@ -327,20 +331,18 @@ JupyterLab 除了可以编辑 Notebook 外，还可以编辑 Markdown 文件，
 
 ## Python 学习资源
 
-Python 语法丰富，模块众多，我们无法也不打算在这个入门教程中介绍 Python 的所有内容。
-读者可以搜索网络资源，自行学习 Python 以及常用的模块。
-
-地震学日常科研中经常使用的 Python 科学计算模块有：
+下面列出一些 Python 以及常用科学计算模块的相关学习资源：
 
+- [Python](https://www.python.org/)
+  - [Python 官方文档](https://docs.python.org/zh-cn/3/)（全面、系统）
+  - [廖雪峰的 Python 教程](https://www.liaoxuefeng.com/wiki/1016959663602400)（较全面、系统）
 - [NumPy](https://numpy.org/)
+  - [NumPy 官方文档](https://numpy.org/doc/stable/)
+  - [A Visual Intro to NumPy and Data Representation](https://jalammar.github.io/visual-numpy/)（简要、形象地介绍 NumPy 数据结构）
 - [Matplotlib](https://matplotlib.org/)
+  - [Matplotlib 官方教程](https://matplotlib.org/stable/tutorials/)
+  - [Scientific Visualization: Python + Matplotlib](https://github.com/rougier/scientific-visualization-book)
 - [pandas](https://pandas.pydata.org/)
+  - [Pandas 官方教程](https://pandas.pydata.org/docs/user_guide/)
 - [SciPy](https://scipy.org/)
-
-下面是一些不错的 Python 学习资源：
-
-- [Python 官方文档](https://docs.python.org/zh-cn/3/)（全面、系统）
-- [Python 教程](https://www.liaoxuefeng.com/wiki/1016959663602400)（较全面、系统）
-- [Scipy Lecture Notes](https://scipy-lectures.org/index.html)（较全面、系统）
```

Comment:
Shall we keep this lecture notes? e.g.,
```
- [SciPy](https://scipy.org/)
  - [SciPy 官方教程](https://docs.scipy.org/doc/scipy/tutorial/)
- 其他综合学习资源
  - [Scipy Lecture Notes](https://scipy-lectures.org/index.html)（较全面、系统）
```

Reply:
https://github.com/scipy-lectures/scipy-lecture-notes/commits/master

This lecture notes are no longer maintained. There are no activities since 2020.